collector/collector.go

```diff
@@ -62,35 +63,29 @@ func NewCollector(factories component.Factories) *Collector {
 	var r io.Reader = f
 	col := &Collector{
 		factories:      factories,
-		parserProvider: parserprovider.NewInMemory(r),
+		mapProvider: parserprovider.NewInMemoryMapProvider(r),
```

Comment:
So should this be `ExpandedMapProvider` instead?

Reply:
```suggestion
		mapProvider: parserprovider.NewExpandMapProvider(parserprovider.NewFileMapProvider(getConfig())),
```

This also allows removing the first part of this function that reads the config into memory since it uses a `FileMapProvider`.

Reply:
Got it, done.